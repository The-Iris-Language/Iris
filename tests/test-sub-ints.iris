.. tests adding ints and assigning to variables or class vars

class Dog () {
    public:
        int age;
}

class Main () {
    public: 
        int univ main() {
            int a = 1 - 2;
            Olympus.print("a = -1");
            a = -1 - 2;
            Olympus.print("a = -3");
            a = -5 - a;
            Olympus.print("a = -8");
            a = -5 - -6;
            Olympus.print("a = 1");
            Dog d;
            d.age = 1 - 1;
            Olympus.print("d.age = 0");
            d.age = -5 - -5;
            Olympus.print("d.age = 0");
            d.age = a - a; 
            Olympus.print("d.age = 0");
            a = d.age - 1;
            Olympus.print("a = -1");
            return 0;
        }
}