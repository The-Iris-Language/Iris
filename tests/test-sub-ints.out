a = -1
a = -3
a = -8
a = 1
d.age = 0
d.age = 0
d.age = 0
a = -1
